docs/community/rust-coding-guidelines/rust-secure-coding-guidelines.md

```diff
@@ -0,0 +1,253 @@
+
+- **BTreeMap:** It maintains its elements in sorted order by their keys.
+- **BinaryHeap:** It maintains its elements in a heap order, which is a complete binary tree where each parent node is less than or equal to its child nodes.
+- **VEC**: It maintains its elements in the order in which they were inserted. ⚠️
+- **LinkedList:** It maintains its elements in the order in which they were inserted. ⚠️
+- **VecDeque:** It maintains its elements in the order in which they were inserted. ⚠️
+
+## Cryptography
+
+### Talk to a cryptographer
+
+If you’re using cryptography in your code and you are **not** a cryptographer, you don’t know what you don’t know (_e.g.,_ small subgroup attacks, malleability attacks, domain-separation attacks, collision resistance versus one-wayness, safe key reuse, correctly using randomness, nonce reuse, CCA security, CPA security, etc). So ask for help from a cryptographer in our team (Alin, Benny, Michael, Rex).
+
+### No Custom Cryptographic Algorithm
+
+Use exclusively the cryptographic primitives exposed by the `aptos-crypto` crate. This does **not** mean you will be able to use them safely, so [talk to a cryptographer](#talk-to-a-cryptographer)!
+
+### Cryptographic Material Management
+
+Adhere strictly to established protocols for generating, storing, and managing cryptographic keys. This includes using secure random sources for key generation, ensuring keys are stored in protected environments, and implementing robust management practices to handle key lifecycle events like rotation and revocation [[Key Management Cheat Sheet]](https://cheatsheetseries.owasp.org/cheatsheets/Key_Management_Cheat_Sheet.html).
+
+### Zeroing Sensitive Data
+
+Use [zeroize](https://docs.rs/zeroize/latest/zeroize/#) for zeroing memory containing sensitive data.
+
+## Misc
+
+### Forget and Memory Leaks
+
+Avoid using `std::mem::forget` in secure development, or any other function that leaks the memory [[forget and memory leaks]](https://anssi-fr.github.io/rust-guide/05_memory.html#forget-and-memory-leaks).
+
+> Reference cycles can also cause memory leakage [[Rustbook: leak]](https://doc.rust-lang.org/book/ch15-06-reference-cycles.html?highlight=leak#reference-cycles-can-leak-memory).
+
+Most memory leaks result in general product reliability problems. If an attacker can intentionally trigger a memory leak, the attacker might be able to launch a denial-of-service attack (by crashing or hanging the program).
+
+### Fuzzing
+Aptos contains harnesses for fuzzing crash-prone code like deserializers, using [`libFuzzer`](https://llvm.org/docs/LibFuzzer.html) through [`cargo fuzz`](https://rust-fuzz.github.io/book/cargo-fuzz.html). For more examples, see the `testsuite/fuzzer` directory where find detailed README.md.
+
+## Appendix
+
+#### Implementation of Comparison Traits
```

Comment:
not sure how relevant this is

Reply:
We have 109+ trait implementation in our codebase, e.g. 27 PartialEq

Reply:
let me rephrase - I think this sample code doesn't provide much value to the overall document, as we don't expect developers copying this when implementing comparison traits. additionally, there are no other code snippets related to other issues.